viewing_party/party.py

```diff
@@ -1,23 +1,191 @@
+    # Loops over each movie in the user's watched movies
+    for movie in range(len(user_data["watched"])):
+        if user_data["watched"][movie] not in user_watched_movies:
+            # if the movie is not in the user_watched_movies, add the movie to user_watched_movies
+            user_watched_movies.append(user_data["watched"][movie])
+    
+    return user_watched_movies
+
+
+def get_unique_watched(user_data):
+    '''
+    Determines which movies the user has watched, but none of their friends have watched.
+
+    Returns:
+    unique_watched_movies (list): Each movie represents a dictionary.
+    '''
+    unique_watched = []
+
+    friends_watched_movies = get_friends_watched_movies(user_data)
+    user_watched_movies = get_user_watched_movies(user_data)
+
+    for movie in user_watched_movies:
+        if (movie not in friends_watched_movies) and (movie not in unique_watched):
+            unique_watched.append(movie)
+    
+    return unique_watched
+    
+
+def get_friends_unique_watched(user_data):
+    '''
+    Determines which movies at least one of the user's friends have watched, but the user has not watched.
+    
+    Returns:
+    friends_unique_watched (list): Each movie represents a dictionary.
+    '''
+    friends_unique_watched = []
+
+    friends_watched_movies = get_friends_watched_movies(user_data)
+    user_watched_movies = get_user_watched_movies(user_data)
+
+    for movie in friends_watched_movies:
+        if (movie not in user_watched_movies) and (movie not in friends_unique_watched):
+            friends_unique_watched.append(movie)
+
+    return friends_unique_watched
 
-        
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
+def get_available_recs(user_data):
+    '''
+    Determines recommended movies if the user has not watched it, the movie is a host of the user, and at least one friend has watched the movie. 
+
+    Returns:
+    available_recs (list): Each movie represents a dictionary.
+    '''
+    available_recs = []
+
+    friends_watched_movies = get_friends_watched_movies(user_data)
+    user_watched_movies = get_user_watched_movies(user_data)
+
+    for movie in friends_watched_movies:
+        if (movie not in user_watched_movies) and (movie["host"] in user_data["subscriptions"]) and (movie not in available_recs):
+            available_recs.append(movie)
+
+    return available_recs
+
 
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
+def get_new_rec_by_genre(user_data):
+    '''
+    Determines recommended movies based on the user's most frequently watched genre.
+
+    Returns:
+    new_rec_by_genre (list): Each movie represents a dictionary.
+    '''
+    friends_watched_movies = get_friends_watched_movies(user_data)
+    user_watched_movies = get_user_watched_movies(user_data)
+    pass

```

Comment:
The `pass` keyword is a placeholder to make sure the code will compile if there are no other statements - we should delete it once we start to write our functions.